recipes/net-snmp/all/conanfile.py

```diff
@@ -0,0 +1,200 @@
+import functools
+import os
+import stat
+from contextlib import contextmanager
+
+from conans import AutoToolsBuildEnvironment, ConanFile, tools
+from conans.errors import ConanInvalidConfiguration
+
+required_conan_version = ">=1.43.0"
+
+
+class NetSnmpConan(ConanFile):
+    name = "net-snmp"
+    description = (
+        "Simple Network Management Protocol (SNMP) is a widely used protocol "
+        "for monitoring the health and welfare of network equipment "
+        "(eg. routers), computer equipment and even devices like UPSs."
+    )
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "http://www.net-snmp.org/"
+    topics = ("snmp",)
+    license = "BSD-3-Clause"
+    settings = ("os", "arch", "compiler", "build_type")
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "with_ipv6": [True, False],
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+        "with_ipv6": True,
+    }
+    requires = ("openssl/1.1.1m",)
+    exports_sources = ("patches/*",)
+
+    @property
+    def _settings_build(self):
+        return getattr(self, "settings_build", self.settings)
+
+    @property
+    def _is_msvc(self):
+        return self._settings_build.compiler in ("Visual Studio", "msvc")
```

Comment:
why settings_build here?

Reply:
I'm not sure actually. I believe I may have copy-pasted this from somewhere without much thought.

Reply:
pretty sure you want `self.settings.compiler` here

Reply:
Before I push a fix, where can I read about the significance of `settings_build` and where it should be used? Also, is it used properly in `validate()`?

Reply:
`settings` refers to profile settings of host profile (target), and `settings_build` to settings of build profile (stuff coming from build requirements).

Reply:
> Can it be cross-built from windows to another OS with another compiler than msvc, from linux to Windows with minGW?

Provided that the `_build_unix()` is invoked, I don't believe there is anything preventing this from being cross-compiled on Windows.

Reply:
So it means that MinGW should work, and there is no reason to prevent Windows build with MinGW on Windows.